File: frontend/src/servicios/productos.ts
/**
 * Cliente de catálogo (GET /productos). Necesario para que la pantalla de venta pueda listar
 * productos; no es una tarea propia de tasks.md, es un detalle de implementación de T037,
 * igual que persistencia/movimientos.py lo fue de T026 en el backend.
 */

import { clienteHttp } from "./clienteHttp";

export interface Producto {
  id_producto: number;
  nombre: string;
  es_granel: boolean;
  lleva_caducidad: boolean;
  precio_efectivo: string;
  moneda: string;
  id_categoria: number | null;
  /** US12: URL externa de imagen para el catálogo de Venta. `null` ⇒ ícono de categoría. */
  url_imagen: string | null;
}

export interface Categoria {
  id_categoria: number;
  nombre: string;
  dias_umbral_inmovilizado: number | null;
  activo: boolean;
}

export function listarProductos(idSucursal?: number): Promise<Producto[]> {
  const query = idSucursal !== undefined ? `?id_sucursal=${idSucursal}` : "";
  return clienteHttp.get<Producto[]>(`/productos${query}`);
}

export function listarCategorias(): Promise<Categoria[]> {
  return clienteHttp.get<Categoria[]>("/categorias");
}

File: backend/rasero/api/productos.py
"""Endpoint GET /productos (T035). Ver servicios/catalogo.py para el porqué de esta adición.

GET /categorias se añade aquí (Parte 3): las pantallas de Venta y Precios necesitan resolver
`id_categoria` -> nombre para elegir el ícono de familia de categoría (La Regla del Ícono por
Categoría, DESIGN.md v1.2.0), sin pasar por el envoltorio de administración.
"""

from fastapi import APIRouter, Depends
from sqlalchemy import select
from sqlalchemy.orm import Session

from rasero.persistencia.modelos import Categoria
from rasero.persistencia.sesion import obtener_sesion
from rasero.servicios import catalogo as servicio_catalogo

router = APIRouter(tags=["productos"])


@router.get("/categorias")
def listar_categorias(
    incluir_inactivas: bool = False, sesion: Session = Depends(obtener_sesion)
) -> list[dict]:
    consulta = select(Categoria).order_by(Categoria.id_categoria)
    if not incluir_inactivas:
        consulta = consulta.where(Categoria.activo.is_(True))
    return [
        {
            "id_categoria": c.id_categoria,
            "nombre": c.nombre,
            "dias_umbral_inmovilizado": c.dias_umbral_inmovilizado,
            "activo": c.activo,
        }
        for c in sesion.execute(consulta).scalars()
    ]


@router.get("/productos")
def listar_productos(
    id_sucursal: int | None = None, sesion: Session = Depends(obtener_sesion)
) -> list[dict]:
    productos = servicio_catalogo.listar_catalogo(sesion, id_sucursal=id_sucursal)
    return [
        {
            "id_producto": p["id_producto"],
            "nombre": p["nombre"],
            "es_granel": p["es_granel"],
            "lleva_caducidad": p["lleva_caducidad"],
            "precio_efectivo": f"{p['precio_efectivo']:.4f}",
            "moneda": p["moneda"],
            "id_categoria": p["id_categoria"],
            "url_imagen": p["url_imagen"],
        }
        for p in productos
    ]
